apollo-federation/src/connectors/json_selection/lit_expr.rs

```diff
@@ -54,13 +55,94 @@ pub(crate) enum LitExpr {
     // of the path, which is never PathList::Empty, because that would mean the
     // LitExpr could stand on its own, using one of the other variants.
     LitPath(WithRange<LitExpr>, WithRange<PathList>),
+
+    // Operator chains: A op B op C ... where all operators are the same type
+    // OpChain contains the operator type and a vector of operands
+    // For example: A ?? B ?? C becomes OpChain(NullCoalescing, [A, B, C])
+    OpChain(LitOp, Vec<WithRange<LitExpr>>),
```

Comment:
```suggestion
    // Operator chains: A op B op C ... where all operators are the same type
    // OpChain contains the operator type and a vector of operands
    // For example: A ?? B ?? C becomes OpChain(NullCoalescing, [A, B, C])
    OpChain(WithRange<LitOp>, Vec<WithRange<LitExpr>>),
```
In order to store source range information for `LitOp` operators, I think you want another `WithRange<_>` wrapper at this level (the `LitExpr::OpChain` enum variant).